test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -340,6 +366,15 @@ public void ParseAssignment ()
                 Assert.Equal(value, kvp.Value);
             };
 
+            testParse("EV_DNE", "", "EV_DNE=#no value just comment");
+            testParse("EV_DNE", "", "EV_DNE= #no value just comment");
+            testParse("EV_DNE", "test", "EV_DNE= test #basic comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a'bc allow singleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a\"bc allow doubleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a$bc allow dollarSign in comment");
+            testParse("EV_DNE", "a'b''c", "EV_DNE=a'b''c #allow inline singleQuotes in unquoted values");
```

Comment:
I'm not sure if these 2 are desirable, now that I reflect on it. Is there a reason you think these should be allowed?

Reply:
It would be very easy to add support to my changes for this feature, I just question its necessity.

Reply:
Of particular note, what would you want for:
```
            testParse("EV_DNE", "a 'b' c", "EV_DNE=a 'b' c #allow inline singleQuotes in unquoted values");
            testParse("EV_DNE", "a \"b\" c", "EV_DNE=a \"b\" c #allow inline doubleQuotes in unquoted values");
```

Should those pass?

If we are ok with these 2 failing, then this covers the examples you gave:
```
        internal static readonly Parser<ValueCalculator> UnquotedValue =
            InterpolatedValue.Or(
                InlineWhitespaceChars.Many().Then(w =>
                    NotControlNorWhitespaceChar("#'\"$").Once().Then(
                        c => NotControlNorWhitespaceChar("$").Many().Select(
                            cs => w.Concat(c).Concat(cs)
                        )
                    )
                ).Text()
                .Select(s => new ValueActual(s))
            ).Many().Select(vs => new ValueCalculator(vs));
```

However, I do wonder if we should support one, or both, or neither of these sets of examples -- with unquoted values. All of these already work with quotes:
```
            testParse("EV_DNE", "a'b''c", "EV_DNE=\"a'b''c\" #this already works");
            testParse("EV_DNE", "a\"b\"\"c", "EV_DNE='a\"b\"\"c' #this already works");
            testParse("EV_DNE", "a 'b' c", "EV_DNE=\"a 'b' c\" #this already works");
            testParse("EV_DNE", "a \"b\" c", "EV_DNE='a \"b\" c' #this already works");
```

Reply:
Quotes within unquoted values is "solved": we will not support that. (see discussion here and in #87)